### PR TITLE
Fixes from review on 42

### PR DESCRIPTION
**Pull request by NiloCK**
Responses to 4 comments from @andrewgordstewart in PR #42.

Recommend reading per-commit; commit messages contain links to original context.

### CODE REVIEW

File: channel/state/state.go [hunk 1]
```diff
@@ -165,3 +166,24 @@ func (s State) RecoverSigner(sig Signature) (types.Address, error) {
 	}
 	return RecoverEthereumMessageSigner(stateHash[:], sig)
 }
+
+// Clone returns a clone of the state
+func (s State) Clone() State {
+	clone := State{}
+
+	// shallow clone of fixed part
```
Comment (andrewgordstewart) [inline channel/state/state.go:174]: Is there an easy way for me to know that the shallow clone here is a safe clone?

File: protocols/direct-funding.go [hunk 2]
```diff
@@ -103,6 +103,8 @@ func NewDirectFundingObjectiveState(initialState state.State, myAddress types.Ad
 			init.MyDepositSafetyThreshold[assetAddress] = threshold
 			init.MyDepositTarget[assetAddress] = myShare.Add(myShare, threshold)
 		}
+	} else {
+		return init, err
```
Comment (andrewgordstewart) [inline protocols/direct-funding.go:107]: Here, I would also prefer the "check errors, then do stuff".
```
channelOutcome, err := outcome.Decode(initialState.VariablePart().EncodedOutcome);
if err != nil {
  // PANIC!
}

...
```

Is there a general guideline for golang developers which addresses this style choice? (I understand the verbosity of my snippet above is one of the pain points of golang.)

**Review by andrewgordstewart — APPROVED** [review]
LGTM!

File: protocols/direct-funding.go (hunk 2 above)
Comment (NiloCK) [reply protocols/direct-funding.go:107]: In this case, there's a more concrete benefit of `if [initialize]; condition { [imperative] } else { [handleError] }`: it limits the scope of the initialized variables to just the `if` block.

Maybe it is better for the (much shorter) error condition to occur first: `if [init]; err != nil { handleError() } else { doStuff() }`

All of this aside, I think the entire condition here can / should be scrapped. More on that in another PR.

File: channel/state/state.go (hunk 1 above)
Comment (NiloCK) [reply channel/state/state.go:174]: These are shallow clones of values that _should never change_.

**Pull request merged**